examples/generic_watcher.rs

```diff
@@ -0,0 +1,39 @@
+// This example shows how to use kube with dynamically known resource kinds.
+
+use color_eyre::Result;
+use futures::prelude::*;
+use kube::{
+    api::{DynamicObject, GroupVersionKind, ListParams, Meta},
+    Api, Client,
+};
+use kube_runtime::{utils::try_flatten_applied, watcher};
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    std::env::set_var("RUST_LOG", "info,kube=debug");
+    env_logger::init();
+
+    // alternatively, you can:
+    // 1) take them CLI arguments
+    // 2) use dynamic discovery apis on kube::Client and e.g. watch all
+    // resources in cluster.
+    let group = std::env::var("GROUP").expect("GROUP not set");
+    let group = group.trim();
+    let version = std::env::var("VERSION").expect("VERSION not set");
+    let version = version.trim();
+    let kind = std::env::var("KIND").expect("KIND not set");
+    let kind = kind.trim();
+
+    let gvk = GroupVersionKind::from_dynamic_gvk(group, version, kind);
+
+    let client = Client::try_default().await?;
+    let api = Api::<DynamicObject>::all_with(client, &gvk);
```

Comment:
The user now has an owned and kind of useless GVK object, but I guess it is still needed for the user to keep around if they want to make a `Controller`.

Reply:
Also, if a user wishes to get rid of this object, they can use e.g. `&{gvk}`. That way `gvk` will be consumed while evaluating the block.